include/oneapi/dpl/internal/dynamic_selection_impl/fixed_resource_policy.h

```diff
@@ -0,0 +1,127 @@
+//
+// Copyright (C) 2023 Intel Corporation
+//
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef _ONEDPL_STATIC_POLICY_IMPL_H
+#define _ONEDPL_STATIC_POLICY_IMPL_H
+#include <vector>
+#include <type_traits>
+#include <memory>
+#include <exception>
+#include "oneapi/dpl/internal/dynamic_selection_impl/scoring_policy_defs.h"
+#if _DS_BACKEND_SYCL != 0
+    #include "oneapi/dpl/internal/dynamic_selection_impl/sycl_backend.h"
+#endif
+namespace oneapi {
+namespace dpl {
+namespace experimental {
+
+#if _DS_BACKEND_SYCL != 0
+  template <typename Backend=sycl_backend>
+#else
+  template <typename Backend>
+#endif
+  struct fixed_resource_policy {
+  private:
+    using backend_t = Backend;
+    using resource_container_t = typename backend_t::resource_container_t;
+    using execution_resource_t = typename backend_t::execution_resource_t;
+    using wrapped_resource_t = typename std::decay_t<Backend>::execution_resource_t;
+
+  public:
+    //policy traits
+    using resource_type = decltype(unwrap(std::declval<wrapped_resource_t>()));
+    using selection_type = oneapi::dpl::experimental::basic_selection_handle_t<fixed_resource_policy<Backend>, execution_resource_t>;
+    using wait_type = typename backend_t::wait_type;
+
+  private:
+    std::shared_ptr<backend_t> backend_;
+
+    struct state_t {
+        resource_container_t resources_;
+        int offset_;
+    };
+
+    std::shared_ptr<state_t> state_;
+
+  public:
+    fixed_resource_policy(int offset=0) {
+        if(offset!=deferred_initialization){
+            initialize(offset);
+        }
+    }
+
+    fixed_resource_policy(const std::vector<resource_type>& u, int offset=0) {
+        if(offset!=deferred_initialization){
+            initialize(u, offset);
+        }
+    }
+
+    auto get_resources()  const {
+      if(backend_){
+          return backend_->get_resources();
+      }else{
+          throw std::runtime_error("Called select before initialization\n");
+      }
+    }
+
+    void initialize(int offset=0) {
+      if(!state_){
+           backend_ = std::make_shared<backend_t>();
+           state_= std::make_shared<state_t>();
+           state_->resources_ = get_resources();
+           state_->offset_ = offset;
+      }
+    }
+
+    void initialize(const std::vector<resource_type>& u, int offset=0) {
+      if(!state_){
+           backend_ = std::make_shared<backend_t>(u);
+           state_= std::make_shared<state_t>();
+           for(auto x : u){
+              state_->resources_.emplace_back(x);
+           }
+           state_->offset_ = offset;
+      }
+    }
+
+    template<typename ...Args>
+    selection_type select(Args&&...) {
+      if(state_){
+          if(!state_->resources_.empty()) {
+              return selection_type{*this, state_->resources_[state_->offset_]};
+          }
+          return selection_type{*this};
```

Comment:
So, if the set of resources is for some reason empty, the returned selection will have a default-constructed resource. Should it try to add it to the resource set then, so that maybe the next time it is not empty?

Reply:
Will address later.